x/metadata/client/cli/cli_test.go

```diff
@@ -3399,3 +3411,114 @@ func (s *IntegrationCLITestSuite) TestWriteSessionCmd() {
 
 	runTxCmdTestCases(s, testCases)
 }
+
+// ---------- tx cmd CountAuthorization tests ----------
+
+func (s *IntegrationCLITestSuite) TestCountAuthorizationIntactTxCommands() {
+	// The scenario being tested is as follows:
+	// There are two owners (1 & 2) and two signers (3 & 4).
+	// 1 & 2 are required signers and 3 & 4 are the actual signers.
+	// It should find a grant (granter: 1 -> grantee: 3) and complain
+	// about 4 not being one of the required signers because grant (granter: 2 -> grantee: 4) does not exist.
```

Comment:
```suggestion
	// There are two owners (1 & 2) and the value owner is also 1.
	// That means that to delete the scope, both 1 and 2 need to be signers.
	// However, we're going to have user 3 sign the `RemoveScopeCmd` calls.
	// We first create the spec, then scope.
	// Then create a `CountAuthorization` grant for granter: 1 -> grantee: 3 with only one use.
	// We attempt a `RemoveScopeCmd` with 3 as the only signer. It should fail because it's not signed by 2.
	// Then we add another `CountAuthorization` grant for granter: 2 -> grantee: 3.
	// We attempt a `RemoveScopeCmd` again with only 3 as the signer. This time it should work.
	//
	// This tests the following:
	// * Makes sure the 1st `CountAuthorization` is not "used" for the failed `TX`. If it were, the 2nd `RemoveScopeCmd` would fail because it's not signed by 1.
	// * Makes sure that authorizations are only used once for the `TX` even if being used for both an owner and the value owner. If that were the case, the 2nd `RemoveScopeCmd` would fail because it wasn't signed by the value owner.
```